wagtail/documents/forms.py

```diff
@@ -61,6 +62,13 @@ def save(self, commit=True):
     class Meta:
         widgets = {"tags": AdminTagWidget, "file": forms.FileInput()}
 
+    def clean_tags(self):
+        tags = self.cleaned_data["tags"]
+        max_tag_length = TagBase._meta.get_field("name").max_length
+        if any(len(tag) > max_tag_length for tag in tags):
+            raise forms.ValidationError("Tag(s) are over 100 characters")
```

Comment:
This error message should reference `max_tag_length` rather than hard-coding the number 100.

Also, this needs tests.

Reply:
And this needs to be translatable too

Reply:
@gasman and @laymonage thanks for the comment. I will work on it

Reply:
I have resolved this